integrations/discord-bot/voxguard/bot.py

```
@@ -0,0 +1,516 @@
+    async def _before_case_expiry(self) -> None:
+        await self.wait_until_ready()
+
+    @tasks.loop(hours=6)
+    async def _retention_sweep(self) -> None:
+        """Apply each guild's retention policy on a schedule."""
+        for guild in self.guilds:
+            cfg = self.runtime.config(guild.id).get("retention", {})
+            try:
+                self.runtime.store.purge_expired(cfg)
+                self.runtime.store.scrub_transcripts(int(cfg.get("transcript_days", 0)))
+            except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)